user/deployment/snaps.md

```diff
@@ -1,38 +1,69 @@
 
-To automate continuous delivery of snaps to the Ubuntu store:
+All it takes is the following in your `.travis.yml`:
 
-1. [Encrypt](https://docs.travis-ci.com/user/encrypting-files) your snapcraft credentials with Travis CI
-2. Add the encrypted credentials to the repository where you have the `snapcraft.yaml` metadata for your package.
-3.  Run `snapcraft` and `snapcraft push` in your `.travis.yml`:
+```yaml
+deploy:
+  provider: snap
+  snap: my.snap
+  channel: edge
+```
+{: data-file=".travis.yml"}
+
+In order for Travis CI to have permission to upload and release the snap on your behalf, you need to export a login token by installing Snapraft:
+
+```bash
+sudo snap install snapcraft --classic
+```
+
+Now export a login covering the ability to push this specific snap, as well as release to the channel you selected (in this example, `edge`):
+
+```bash
+snap export-login --snaps my-snap-name --channels edge -
+```
 
-    ```yaml
-    deploy:
-      skip_cleanup: true
-      provider: script
-      script: docker run -v $(pwd):$(pwd) -w $(pwd) snapcore/snapcraft sh -c "apt update && snapcraft && snapcraft push *.snap --release edge"
-      on:
-        branch: master
-    ```
+(note the final `-`, which requests the login be exported to stdout instead of a file.) Copy the token, and put it into the `$SNAP_TOKEN` environment variable in Travis CI:
 
-The `snapcraft enable-ci travis` command will assist you getting the credential, encrypting it and adding the right script to the deploy section of your `.travis.yml` file. Here is a [tutorial that will guide you setting up the continuous delivery from Travis CI](https://tutorials.ubuntu.com/tutorial/continuous-snap-delivery-from-travis-ci#0) on your project.
+```bash
+travis env set SNAP_TOKEN "<token>"
+```
+
+If you use a different environment variable, you'll need to specify the `token` option for the provider:
+
+```yaml
+deploy:
+  provider: snap
+  snap: my.snap
+  channel: edge
+  token: $MY_SNAP_TOKEN
+```
+{: data-file=".travis.yml"}
 
-Note that the `edge` channel is intended for crowdtesting with your community of early adopters. With this deployment script in Travis, every time a pull request lands into the master branch, a new snap will be published to edge. Your testers can install it in any of the supported Linux distros with:
+This example is almost certainly not ideal, as you probably want to upload a snap built during CI. Set `skip_cleanup` to `true` to prevent Travis CI from deleting it before the `deploy` step.
+
+```yaml
+deploy:
+  provider: snap
+  snap: my.snap
+  channel: edge
+  skip_cleanup: true
+```
+{: data-file=".travis.yml"}
+
+Note that the `edge` channel is intended for crowdtesting with your community of early adopters. With this deployment setup in Travis, every time a pull request lands, a new snap will be published to edge. Your testers can install it in any of the supported Linux distros with:
```

Comment:
Note: The `edge` channel is intended for the bleeding edge: your every commit to master will be built and uploaded. This is great for crowd-testing with your community of early adopters, who can install your app in any of the \[supported Linux distributions](https://docs.snapcraft.io/core/install) with:
```bash
sudo snap install my-snap-name --edge
```

Each upload gets a monotonically increasing integer. When you're ready, you can release one of these built commits to the stable channel for the public to discover in the \[Snap storefront](https://snapcraft.io/store). For example, you could promote the very first upload to stable:
```bash
snapcraft release my-snap-name 1 edge
```

Reply:
Done.